src/bin/pg_autoctl/pgsql.c

```diff
@@ -831,6 +840,101 @@ pgAutoCtlDebugNoticeProcessor(void *arg, const char *message)
 }
 
 
+/*
+ * pgsql_begin is responsible for opening a mutli statement connection and
+ * opening a transaction block by issuing a 'BEGIN' query.
+ */
+bool
+pgsql_begin(PGSQL *pgsql)
+{
+	PGconn *connection;
+
+	pgsql->connectionStatementType = PGSQL_CONNECTION_MULTI_STATEMENT;
+	connection = pgsql_open_connection(pgsql);
+	if (connection == NULL)
```

Comment:
Could we check for "transaction already in progress" errors and report it as a BUG, possibly forcing an exit?